src/protocol/mod.rs

```diff
@@ -69,6 +70,2044 @@ pub mod xv;
+    #[cfg(feature = "shape")]
+    ShapeRectangles(shape::RectanglesRequest<'input>),
+    #[cfg(feature = "shape")]
+    ShapeMask(shape::MaskRequest),
+    #[cfg(feature = "shape")]
+    ShapeCombine(shape::CombineRequest),
+    #[cfg(feature = "shape")]
+    ShapeOffset(shape::OffsetRequest),
+    #[cfg(feature = "shape")]
+    ShapeQueryExtents(shape::QueryExtentsRequest),
+    #[cfg(feature = "shape")]
+    ShapeSelectInput(shape::SelectInputRequest),
+    #[cfg(feature = "shape")]
+    ShapeInputSelected(shape::InputSelectedRequest),
+    #[cfg(feature = "shape")]
+    ShapeGetRectangles(shape::GetRectanglesRequest),
+    #[cfg(feature = "shm")]
+    ShmQueryVersion(shm::QueryVersionRequest),
+    #[cfg(feature = "shm")]
+    ShmAttach(shm::AttachRequest),
+    #[cfg(feature = "shm")]
+    ShmDetach(shm::DetachRequest),
+    #[cfg(feature = "shm")]
+    ShmPutImage(shm::PutImageRequest),
+    #[cfg(feature = "shm")]
+    ShmGetImage(shm::GetImageRequest),
+    #[cfg(feature = "shm")]
+    ShmCreatePixmap(shm::CreatePixmapRequest),
+    #[cfg(feature = "shm")]
+    ShmAttachFd(shm::AttachFdRequest),
+    #[cfg(feature = "shm")]
+    ShmCreateSegment(shm::CreateSegmentRequest),
+    #[cfg(feature = "sync")]
+    SyncInitialize(sync::InitializeRequest),
+    #[cfg(feature = "sync")]
+    SyncListSystemCounters(sync::ListSystemCountersRequest),
+    #[cfg(feature = "sync")]
+    SyncCreateCounter(sync::CreateCounterRequest),
+    #[cfg(feature = "sync")]
+    SyncDestroyCounter(sync::DestroyCounterRequest),
+    #[cfg(feature = "sync")]
+    SyncQueryCounter(sync::QueryCounterRequest),
+    #[cfg(feature = "sync")]
+    SyncAwait(sync::AwaitRequest<'input>),
+    #[cfg(feature = "sync")]
+    SyncChangeCounter(sync::ChangeCounterRequest),
+    #[cfg(feature = "sync")]
+    SyncSetCounter(sync::SetCounterRequest),
+    #[cfg(feature = "sync")]
+    SyncCreateAlarm(sync::CreateAlarmRequest<'input>),
+    #[cfg(feature = "sync")]
+    SyncChangeAlarm(sync::ChangeAlarmRequest<'input>),
+    #[cfg(feature = "sync")]
+    SyncDestroyAlarm(sync::DestroyAlarmRequest),
+    #[cfg(feature = "sync")]
+    SyncQueryAlarm(sync::QueryAlarmRequest),
+    #[cfg(feature = "sync")]
+    SyncSetPriority(sync::SetPriorityRequest),
+    #[cfg(feature = "sync")]
+    SyncGetPriority(sync::GetPriorityRequest),
+    #[cfg(feature = "sync")]
+    SyncCreateFence(sync::CreateFenceRequest),
+    #[cfg(feature = "sync")]
+    SyncTriggerFence(sync::TriggerFenceRequest),
+    #[cfg(feature = "sync")]
+    SyncResetFence(sync::ResetFenceRequest),
+    #[cfg(feature = "sync")]
+    SyncDestroyFence(sync::DestroyFenceRequest),
+    #[cfg(feature = "sync")]
+    SyncQueryFence(sync::QueryFenceRequest),
+    #[cfg(feature = "sync")]
+    SyncAwaitFence(sync::AwaitFenceRequest<'input>),
+    Xc_miscGetVersion(xc_misc::GetVersionRequest),
+    Xc_miscGetXIDRange(xc_misc::GetXIDRangeRequest),
+    Xc_miscGetXIDList(xc_misc::GetXIDListRequest),
```

Comment:
This surely needs to be fixed. Maybe doing something similar to `to_rust_type_name` in `get_ns_name_prefix`?